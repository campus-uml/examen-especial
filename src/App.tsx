const App = () => {
  return (
    <div className="max-w-md md:max-w-2xl lg:max-w-4xl mx-auto p-4">
      <h1 className="text-2xl sm:text-3xl font-bold text-center mb-6">Hotels</h1>
      <div className="flex flex-col gap-4">
        {/* Hotel 1 */}
        <div className="flex items-center gap-4 p-4 border rounded-lg shadow-sm">
          <img src="./img/img1.jpg" alt="Ledadu Beach" className="w-16 h-16 md:w-20 md:h-20 rounded-md" />
          <div className="flex flex-col flex-grow">
            <h2 className="text-lg font-semibold">Ledadu Beach</h2>
            <p className="text-sm text-gray-600">3 days 2 nights</p>
            <p className="text-sm text-gray-500">📍 Australia</p>
          </div>
          <p className="text-lg font-bold">$20<span className="text-sm">/Person</span></p>
        </div>

        {/* Hotel 2 */}
        <div className="flex items-center gap-4 p-4 border rounded-lg shadow-sm">
          <img src="./img/img2.jpg" alt="Endigada Beach" className="w-16 h-16 md:w-20 md:h-20 rounded-md" />
          <div className="flex flex-col flex-grow">
            <h2 className="text-lg font-semibold">Endigada Beach</h2>
            <p className="text-sm text-gray-600">5 days 4 nights</p>
            <p className="text-sm text-gray-500">📍Japon</p>
          </div>
          <p className="text-lg font-bold">$18<span className="text-sm">/Person</span></p>
        </div>

        {/* Hotel 3 */}
        <div className="flex items-center gap-4 p-4 border rounded-lg shadow-sm">
          <img src="./img/img3.jpg" alt="Doreen Tower" className="w-16 h-16 md:w-20 md:h-20 rounded-md" />
          <div className="flex flex-col flex-grow">
            <h2 className="text-lg font-semibold">Doreen Tower</h2>
            <p className="text-sm text-gray-600">5 days 4 nights</p>
            <p className="text-sm text-gray-500">📍 USA</p>
          </div>
          <p className="text-lg font-bold">$14<span className="text-sm">/Person</span></p>
        </div>

        {/* Hotel 4 */}
        <div className="flex items-center gap-4 p-4 border rounded-lg shadow-sm">
          <img src="./img/img4.jpg" alt="Royal Palace" className="w-16 h-16 md:w-20 md:h-20 rounded-md" />
          <div className="flex flex-col flex-grow">
            <h2 className="text-lg font-semibold">Royal Palace</h2>
            <p className="text-sm text-gray-600">5 days 4 nights</p>
            <p className="text-sm text-gray-500">📍 China</p>
          </div>
          <p className="text-lg font-bold">$21<span className="text-sm">/Person</span></p>
        </div>

          {/* Hotel 5 */}
            <div className="flex items-center gap-4 p-4 border rounded-lg shadow-sm">
          <img src="./img/img5.jpg" alt="Royal Palace" className="w-16 h-16 md:w-20 md:h-20 rounded-md" />
          <div className="flex flex-col flex-grow">
            <h2 className="text-lg font-semibold">El Dorado</h2>
            <p className="text-sm text-gray-600">5 days 4 nights</p>
            <p className="text-sm text-gray-500">📍España</p>
          </div>
          <p className="text-lg font-bold">$26<span className="text-sm">/Person</span></p>
        </div>

          {/* Hotel 6 */}
          <div className="flex items-center gap-4 p-4 border rounded-lg shadow-sm">
          <img src="./img/img6.jpg" alt="Royal Palace" className="w-16 h-16 md:w-20 md:h-20 rounded-md" />
          <div className="flex flex-col flex-grow">
            <h2 className="text-lg font-semibold">San Marino</h2>
            <p className="text-sm text-gray-600">5 days 4 nights</p>
            <p className="text-sm text-gray-500">📍Argentina</p>
          </div>
          <p className="text-lg font-bold">$28<span className="text-sm">/Person</span></p>
        </div>
      </div>
    </div>
  );
};

export default App;