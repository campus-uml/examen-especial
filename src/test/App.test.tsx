
import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import App from "../App";

describe("Testing de App", () => {
  it("Debe renderizarse correctamente", () => {
    const { baseElement } = render(<App />);
    expect(baseElement).toMatchSnapshot();
  });

  it('Debe mostrar el título "Hotels"', () => {
    const { getByText } = render(<App />);
    expect(getByText("Hotels")).toBeInTheDocument();
  });

  it("Debe mostrar la lista de hoteles con sus nombres y ubicaciones", () => {
    const { getByText } = render(<App />);
    const hotels = [
      { name: "Ledadu Beach", location: "📍 Australia" },
      { name: "Endigada Beach", location: "📍Japon" },
      { name: "Doreen Tower", location: "📍 USA" },
      { name: "Royal Palace", location: "📍 China" },
      { name: "El Dorado", location: "📍España" },
      { name: "San Marino", location: "📍Argentina" },
    ];

    hotels.forEach((hotel) => {
      expect(getByText(hotel.name)).toBeInTheDocument();
      expect(getByText(hotel.location)).toBeInTheDocument();
    });
  });

  it("Debe mostrar imágenes para cada hotel", () => {
    const { getAllByRole } = render(<App />);
    const images = getAllByRole("img");
    expect(images).toHaveLength(6);

    images.forEach((img) => {
      expect(img).toBeInTheDocument();
    });
  });
});