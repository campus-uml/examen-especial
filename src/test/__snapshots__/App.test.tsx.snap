// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing de App > Debe renderizarse correctamente 1`] = `
<body>
  <div>
    <div
      class="max-w-md md:max-w-2xl lg:max-w-4xl mx-auto p-4"
    >
      <h1
        class="text-2xl sm:text-3xl font-bold text-center mb-6"
      >
        Hotels
      </h1>
      <div
        class="flex flex-col gap-4"
      >
        <div
          class="flex items-center gap-4 p-4 border rounded-lg shadow-sm"
          data-testid="hotel-item"
        >
          <img
            alt="Ledadu Beach"
            class="w-16 h-16 md:w-20 md:h-20 rounded-md"
            src="./img/img1.jpg"
          />
          <div
            class="flex flex-col flex-grow"
          >
            <h2
              class="text-lg font-semibold"
              data-testid="hotel-name"
            >
              Ledadu Beach
            </h2>
            <p
              class="text-sm text-gray-600"
            >
              3 days 2 nights
            </p>
            <p
              class="text-sm text-gray-500"
            >
              üìç Australia
            </p>
          </div>
          <p
            class="text-lg font-bold"
          >
            $20
            <span
              class="text-sm"
            >
              /Person
            </span>
          </p>
        </div>
        <div
          class="flex items-center gap-4 p-4 border rounded-lg shadow-sm"
          data-testid="hotel-item"
        >
          <img
            alt="Endigada Beach"
            class="w-16 h-16 md:w-20 md:h-20 rounded-md"
            src="./img/img2.jpg"
          />
          <div
            class="flex flex-col flex-grow"
          >
            <h2
              class="text-lg font-semibold"
              data-testid="hotel-name"
            >
              Endigada Beach
            </h2>
            <p
              class="text-sm text-gray-600"
            >
              5 days 4 nights
            </p>
            <p
              class="text-sm text-gray-500"
            >
              üìçJapon
            </p>
          </div>
          <p
            class="text-lg font-bold"
          >
            $18
            <span
              class="text-sm"
            >
              /Person
            </span>
          </p>
        </div>
        <div
          class="flex items-center gap-4 p-4 border rounded-lg shadow-sm"
          data-testid="hotel-item"
        >
          <img
            alt="Doreen Tower"
            class="w-16 h-16 md:w-20 md:h-20 rounded-md"
            src="./img/img3.jpg"
          />
          <div
            class="flex flex-col flex-grow"
          >
            <h2
              class="text-lg font-semibold"
              data-testid="hotel-name"
            >
              Doreen Tower
            </h2>
            <p
              class="text-sm text-gray-600"
            >
              5 days 4 nights
            </p>
            <p
              class="text-sm text-gray-500"
            >
              üìç USA
            </p>
          </div>
          <p
            class="text-lg font-bold"
          >
            $14
            <span
              class="text-sm"
            >
              /Person
            </span>
          </p>
        </div>
        <div
          class="flex items-center gap-4 p-4 border rounded-lg shadow-sm"
          data-testid="hotel-item"
        >
          <img
            alt="Royal Palace"
            class="w-16 h-16 md:w-20 md:h-20 rounded-md"
            src="./img/img4.jpg"
          />
          <div
            class="flex flex-col flex-grow"
          >
            <h2
              class="text-lg font-semibold"
              data-testid="hotel-name"
            >
              Royal Palace
            </h2>
            <p
              class="text-sm text-gray-600"
            >
              5 days 4 nights
            </p>
            <p
              class="text-sm text-gray-500"
            >
              üìç China
            </p>
          </div>
          <p
            class="text-lg font-bold"
          >
            $21
            <span
              class="text-sm"
            >
              /Person
            </span>
          </p>
        </div>
        <div
          class="flex items-center gap-4 p-4 border rounded-lg shadow-sm"
          data-testid="hotel-item"
        >
          <img
            alt="Royal Palace"
            class="w-16 h-16 md:w-20 md:h-20 rounded-md"
            src="./img/img5.jpg"
          />
          <div
            class="flex flex-col flex-grow"
          >
            <h2
              class="text-lg font-semibold"
              data-testid="hotel-name"
            >
              El Dorado
            </h2>
            <p
              class="text-sm text-gray-600"
            >
              5 days 4 nights
            </p>
            <p
              class="text-sm text-gray-500"
            >
              üìçEspa√±a
            </p>
          </div>
          <p
            class="text-lg font-bold"
          >
            $26
            <span
              class="text-sm"
            >
              /Person
            </span>
          </p>
        </div>
        <div
          class="flex items-center gap-4 p-4 border rounded-lg shadow-sm"
          data-testid="hotel-item"
        >
          <img
            alt="Royal Palace"
            class="w-16 h-16 md:w-20 md:h-20 rounded-md"
            src="./img/img6.jpg"
          />
          <div
            class="flex flex-col flex-grow"
          >
            <h2
              class="text-lg font-semibold"
              data-testid="hotel-name"
            >
              San Marino
            </h2>
            <p
              class="text-sm text-gray-600"
            >
              5 days 4 nights
            </p>
            <p
              class="text-sm text-gray-500"
            >
              üìçArgentina
            </p>
          </div>
          <p
            class="text-lg font-bold"
          >
            $28
            <span
              class="text-sm"
            >
              /Person
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>
</body>
`;
